.circleci/config.yml

```diff
@@ -1224,6 +1242,15 @@ jobs:
             # lsof is used by Colony in its stop-blockchain-client.sh script
             sudo apt update
             sudo apt-get --quiet --assume-yes --no-install-recommends install lsof
+      - when:
+          condition: true
+          <<: *steps_install_python_dependencies
+      - when:
+          condition:
+            matches:
+                pattern: /^.+rust.+$/
+                value: << parameters.image >>
+          <<: *steps_install_foundry
```

Comment:
If you add caching as I suggested above, you could just install it unconditionally and make this much simpler.